packages/web3-eth-abi/test/unit/types.test.ts

```diff
@@ -0,0 +1,151 @@
+import { Address, Bytes, Numbers } from 'web3-utils';
+import { expectTypeOf, typecheck } from '@humeris/espresso-shot';
+import { MatchPrimitiveType } from '../../src/types';
+
+describe('types', () => {
+	describe('primitive types', () => {
+		describe('bool', () => {
+			typecheck('should extend the boolean type', () => {
+				const b: MatchPrimitiveType<'bool', []> = true;
+
+				return expectTypeOf(b).toExtend<boolean>();
+			});
+
+			typecheck('should extend the boolean type array', () => {
+				const b: MatchPrimitiveType<'bool[]', []> = [true];
+
+				return expectTypeOf(b).toExtend<boolean[]>();
+			});
+
+			typecheck('should extend the boolean type fixed array', () => {
+				const b: MatchPrimitiveType<'bool[3]', []> = [true, true, true];
+
+				return expectTypeOf(b).toExtend<[boolean, boolean, boolean]>();
+			});
+		});
+
+		describe('string', () => {
+			typecheck('should extend the string type', () => {
+				const b: MatchPrimitiveType<'string', []> = 'a';
+
+				return expectTypeOf(b).toExtend<string>();
+			});
+
+			typecheck('should extend the string type array', () => {
+				const b: MatchPrimitiveType<'string[]', []> = ['a'];
+
+				return expectTypeOf(b).toExtend<string[]>();
+			});
+
+			typecheck('should extend the string type fixed array', () => {
+				const b: MatchPrimitiveType<'string[3]', []> = ['a', 'a', 'a'];
+
+				return expectTypeOf(b).toExtend<[string, string, string]>();
+			});
+		});
+
+		describe('address', () => {
+			typecheck('should extend correct type', () =>
+				expectTypeOf<MatchPrimitiveType<'address', []>>().toExtend<Address>(),
```

Comment:
Why switch to this style of tests when the above use `const b: MatchPrimitiveType<...> = ...;` format?

Reply:
Fixed that feedback by using consistent pattern.